src/DigitalSubscriberAppBanner.tsx

```diff
@@ -283,19 +286,45 @@ export const storeIcon = css`
     }
 `;
 
+const withTryCatchBehaviour = (fn: () => void): void => {
+    try {
+        fn();
+    } catch (e) {}
+};
+
 export const DigitalSubscriberAppBanner: React.FC<Props> = ({
-    onButtonClick,
+    logButtonClickWithBraze,
+    submitComponentEvent,
     brazeMessageProps: { header, body },
 }: Props) => {
     const [showBanner, setShowBanner] = useState(true);
 
     const onCloseClick = (
         evt: React.MouseEvent<HTMLButtonElement, MouseEvent>,
-        buttonId: number,
+        internalButtonId: number,
     ): void => {
         evt.preventDefault();
-        onButtonClick(buttonId);
+
         setShowBanner(false);
+
+        withTryCatchBehaviour(() => {
```

Comment:
I'm not sure that the indirection of the `withTryCatchBehaviour` function is really giving us anything here. Is there a reason not to use try / catch directly here and below? This might also give us the opportunity to make better local decisions about how to recover from errors instead of silently discarding them as we do now.

Reply:
I think if we switch to logging the errors, which for now would be the common way of handling them, then I can just about justify the existence of this function. I don't feel strongly though, it was mostly just that `try/catch` repeated looked a little untidy to me as it adds so many extra lines!

Reply:
I wonder if a `finally` block would be a better fit to ensure that we can perform some logic whether or not the handler fails?

Even re-throwing the error should be fine once we've done our cleanup work, so that sentry handles it as normal.